Source/Filter/NCIGodfreyFilter.cpp

```diff
@@ -0,0 +1,86 @@
+#include <WarpX.H>
+#include <NCIGodfreyFilter.H>
+#include <NCIGodfreyTables.H>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+using namespace amrex;
+
+NCIGodfreyFilter::NCIGodfreyFilter(godfrey_coeff_set coeff_set_, amrex::Real cdtodz_, amrex::Real l_lower_order_in_v_){
+    // Store parameters into class data members
+    coeff_set = coeff_set_;
+    cdtodz = cdtodz_;
+    l_lower_order_in_v = l_lower_order_in_v_;
+    // NCI Godfrey filter has fixed size, and is applied along z only.
+#if (AMREX_SPACEDIM == 3)
+    stencil_length_each_dir = {1,1,5};
+    slen = {1,1,5};
+#else
+    stencil_length_each_dir = {1,5};
+    slen = {1,5,1};
+#endif    
+}
+
+void NCIGodfreyFilter::ComputeStencils(){
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+#if ( AMREX_SPACEDIM == 3 )
+        slen.z==5,
+#else
+        slen.y==5, 
+#endif
+        "ERROR: NCI filter requires 5 points in z");
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(l_lower_order_in_v==1, 
+        "ERROR: NCI corrector requires l_lower_order_in_v=1, i.e., Galerkin scheme");
+
+    // Godfrey filter uses a table to compute the filter coefficients.
+    // Get the appropriate table (one for Ex, Ey and Bz, the other 
+    // one for Bx, By, Ez).
+    Real table_nci[tab_length][tab_width];
+    if        (coeff_set == godfrey_coeff_set::Ex_Ey_Bz) {
+        std::copy(&table_nci_godfrey_Ex_Ey_Bz[0][0], 
+                  &table_nci_godfrey_Ex_Ey_Bz[0][0]+tab_width*tab_length, 
+                  &table_nci[0][0]);
```

Comment:
@WeiqunZhang I am not too sure about the scope of variable `table_nci`, as well as `table_nci_godfrey_Ex_Ey_Bz ` and `table_nci_godfrey_Bx_By_Ez ` in `Source/Utils/NCIGodfreyTables.H`. They should be deleted when exiting `NCIGodfreyFilter::ComputeStencils()`, right? There is no reason to keep the tables throughout the simulation.

Reply:
`table_nci` will be gone after the constructor finishes.  We don't have to do anything.  But I also think there are no reasons to use `table_nci` at all.  It is used in only one place, we could simply use `if` there.